gateway/src/test/java/io/camunda/zeebe/gateway/admin/exporting/ExportingControlServiceTest.java

```diff
@@ -0,0 +1,200 @@
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.gateway.admin.exporting;
+
+import static io.camunda.zeebe.gateway.admin.exporting.ExportingControlServiceTest.RequestMatcher.requestTo;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.junit.jupiter.api.Named.named;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.gateway.admin.exporting.ExportingControlService.InvalidTopologyException;
+import io.camunda.zeebe.gateway.impl.broker.BrokerClient;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerClusterState;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerClusterStateImpl;
+import io.camunda.zeebe.gateway.impl.broker.cluster.BrokerTopologyManager;
+import io.camunda.zeebe.gateway.impl.broker.request.BrokerRequest;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentMatcher;
+
+public class ExportingControlServiceTest {
+
+  @ParameterizedTest
+  @MethodSource("validTopologies")
+  void shouldPauseOnAllBrokersAndPartitions(final BrokerClusterState topology) {
+    // given
+    final var client = setupBrokerClient(topology);
+    final var service = new ExportingControlService(client);
+
+    // when
+    service.pauseExporting().join();
+
+    // then
+    for (final var partition : topology.getPartitions()) {
+      for (final var follower :
+          Optional.ofNullable(topology.getFollowersForPartition(partition)).orElse(Set.of())) {
+        verify(client).sendRequest(requestTo(partition, follower));
+      }
+      verify(client).sendRequest(requestTo(partition, topology.getLeaderForPartition(partition)));
+
+      for (final var inactive :
+          Optional.ofNullable(topology.getInactiveNodesForPartition(partition)).orElse(Set.of())) {
+        verify(client, never()).sendRequest(requestTo(partition, inactive));
+      }
```

Comment:
:x: I think we should also pause on inactive partitions. Inactive partitions are temporarily unhealthy ones. So they should be paused to ensure that once they recover from the error (after a restart), they do not unpause.

Reply:
Thanks, should be fixed now :+1: